dcm4che-imageio/src/main/java/org/dcm4che3/imageio/stream/ByteArrayImageInputStream.java

```diff
@@ -0,0 +1,51 @@
+//
+///////////////////////////////////////////////////////////////
+//                C O P Y R I G H T  (c) 2019                //
+//        Agfa HealthCare N.V. and/or its affiliates         //
+//                    All Rights Reserved                    //
+///////////////////////////////////////////////////////////////
+//                                                           //
+//       THIS IS UNPUBLISHED PROPRIETARY SOURCE CODE OF      //
+//        Agfa HealthCare N.V. and/or its affiliates.        //
+//      The copyright notice above does not evidence any     //
+//     actual or intended publication of such source code.   //
+//                                                           //
+///////////////////////////////////////////////////////////////
+//
+package org.dcm4che3.imageio.stream;
+
+import javax.imageio.stream.ImageInputStreamImpl;
+import java.io.IOException;
+
+/**
+ * ImageInputStream which wraps an in-memory byte[]
+ * @author Andrew Cowan (andrew.cowan@agfa.com)
+ */
```

Comment:
Efficient IIS implementation for byte[].  You can do the same thing with an InputStream, but this will allow random access to the byte[] that is already in memory:  if we already have the data in memory, we might as well use it.